backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != "UNK"
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $hash = hash
+    }
+
+***
+get all data
+***
+    SELECT vid,variant.variantanswer AS variantanswer,useranswer,param,cid,vers,quiz FROM userAnswer LEFT JOIN variant ON userAnswer.variant=variant.vid WHERE hash=hash;
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #8 (rows 327-334)
+super-view data from useranswer on moment
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != "UNK"
+    $rows = NULL
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $moment = moment
+    }
+
+***
+get all data
+***
+    SELECT vid,variant.variantanswer AS variantanswer,useranswer,param,cid,vers,quiz FROM userAnswer LEFT JOIN variant ON userAnswer.variant=variant.vid WHERE moment=moment;
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #9 - 345 - 351
+super-view data on quizname
+
+***
+Pre-req: 
+    isSuperUser = TRUE
+    $hash != “UNK”
+***
+
+***
+* send parameters input to showDuggaservice.php
+***
+    send
+    {
+        $moment = moment
+    }
+
+***
+get all entrynames
+***
+    SELECT entryname FROM listentries WHERE lid=moment;
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #10 - 374 - 402
+view data on userAnswer
+
+***
+Pre-req: !isSuperUser 
+	  $hash != “UNK” && password != “UNK”
+***
+    send
+    {
+        $cid = 2
+        $coursevers = 52432
+        $duggaid = 1
+        $moment = 2002
+        $variant = 3
+        $hash = “ghj1ghj2”
+        $haspwd = “asddasdd”
+        $answer = NULL
+    }
+
+    SQL-query INSERT INTO userAnswer (cid,quiz,moment,vers,variant,hash,password,timesSubmitted,timesAccessed,useranswer,submitted) VALUES(2, 1, 2002, 52432 , 3, “ghj1ghj2” , “asddasdd”,1 ,1 , NULL ,now());")
```

Comment:
i executed this query in mysql and tried to find the results in LenaSYS, i could not find it there.
did you test this to see if the values being sent actually appear in LenaSYS, if yes, the could you tell med where to find the results, i dont know if im searching in the wrong place..

Reply:
We have only been using DB and seeing if the data actually works correctly. Don't think we have to try LenaSYS yet since the code implementation might be incorrect at the moment.